### PR TITLE
feat(datepicker): esc can close datepicker

### LINKED ISSUE #3890: Datepicker: does not always close by 'Esc' key
Go to the demo page and open the first datepicker (basic example). Press 'esc' - it closes. Now open again and click on the year/month buttons. Press 'esc' - it does not close.

### PR DESCRIPTION
**Pull request by 5earle**
Closes #3890

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

**Comment by 5earle** [comment]
Addresses #3890

File: src/component-loader/component-loader.class.ts [hunk 1]
```diff
@@ -298,6 +301,16 @@ export class ComponentLoader<T> {
         });
       });
     }
+    if (this._listenOpts.outsideEsc) {
+      const target = this._componentRef.location.nativeElement;
+      setTimeout(() => {
```
Comment (valorkin) [inline src/component-loader/component-loader.class.ts]: why setTimeout is required?

**Review by valorkin — CHANGES_REQUESTED** [review]
Hey, thanks for PR!
I think we could merge close on ESC
but not sure about tab related logic

if you could split this PR into 2 (one with ESC one with Tab, logic) it would be easier to merge

**Comment by Domainv** [comment]
'tab can close datepicker' functionality was cut out as unnecessary for now

**Review by Domainv — DISMISSED** [review]

**Comment by ludmilanesvitiy** [comment]
Smoke tested in Chrome, FF  -Esc always close the Datepicker. Looks good.

**Review by valorkin — APPROVED** [review]
LGTM, thanks ;)

**Pull request merged**